.travis.yml

```diff
@@ -13,34 +17,51 @@ jdk: oraclejdk8
 env:
   matrix:
     # ECLIPSE_TARGET is the targeted platform: mars is the default target
-    # JAVA_RUNTIME specifies the execution runtime for tests; must match an <id> in .travis-toolchains.xml
+    # JAVA_RUNTIME specifies the execution runtime for tests; must match an <id> in .travisci/toolchains.xml
     # SWT_GTK3=0 to force GTK2 on Mars as its GTK3 support is buggy
     - JAVA_RUNTIME=JavaSE-1.7 SWT_GTK3=0
     - JAVA_RUNTIME=JavaSE-1.8 ECLIPSE_TARGET=neon
     - JAVA_RUNTIME=JavaSE-1.8 ECLIPSE_TARGET=oxygen MAVEN_FLAGS=-Pjacoco
   global:
-    - PATH=$PWD/google-cloud-sdk/bin:$PATH
+    - GCLOUDSDK=$HOME/google-cloud-sdk
+    - PATH=$GCLOUDSDK/bin:$PATH
     # Our maven build typically reports about 512m
     - MAVEN_OPTS=-Xmx700m
     - CLOUDSDK_CORE_DISABLE_USAGE_REPORTING=true
     - DISPLAY=:99.0
+    - GCSBUILDBUCKET=gs://gcloud-eclipse-testing
 before_script:
   - "sh -e /etc/init.d/xvfb start"
   - sleep 3 # give xvfb some time to start
   - metacity --sm-disable --replace &
   - sleep 3 # give metacity some time to start
 script: ./mvnw -V -B -Ptravis --fail-at-end verify ${MAVEN_FLAGS}
      ${ECLIPSE_TARGET:+-Declipse.target=${ECLIPSE_TARGET}}
-     --toolchains=.travis-toolchains.xml
+     --toolchains=.travisci/toolchains.xml
      -Dtoolchain.java.runtime=${JAVA_RUNTIME}
-addons:
-  apt:
-    packages:
-      - metacity
 cache:
   directories:
    - $HOME/.m2
-   - $HOME/google-cloud-sdk
+   - $GCLOUDSDK
 after_success:
   # test req'd as we don't run coverage on all elements of the matrix
   - if [ -d build/jacoco/target ]; then bash <(curl -s https://codecov.io/bash); fi
+after_script:
+  # Upload build reports to our Google Cloud Storage bucket
+  - openssl aes-256-cbc -K $encrypted_42b011b17512_key -iv $encrypted_42b011b17512_iv
+      -in .travisci/travis-service-account.json.enc -out /tmp/travis-service-account.json -d
+  - $GCLOUDSDK/bin/gcloud auth activate-service-account --key-file=/tmp/travis-service-account.json
+  # Copy surefire-reports and screenshots
+  - mkdir build-results && .travisci/copy-test-results.sh plugins build-results
+  - $GCLOUDSDK/bin/gsutil -m -q
+      cp -r -z 'txt,xml,html' -a public-read
+        build-results/*
+        $GCSBUILDBUCKET/travis/${TRAVIS_BUILD_NUMBER}/${TRAVIS_JOB_NUMBER}-${TRAVIS_OS_NAME}/
+  # Copy build results (if successul; for only one element of build matrix)
+  - if [ -d gcp-repo/target/repository -a -d build/jacoco/target ]; then
+      $GCLOUDSDK/bin/gsutil -m -q
+        cp -r -z 'txt,xml,html' -a public-read
```

Comment:
File extensions don't seem right.

Reply:
How so?  These are the extensions for `gsutil` to compress.  We have the following extensions in what we upload:

  - html: a stub in our `gcp-repo` that redirects to http://cloud.google.com/eclipse
  - index: the `p2.index` file, which is tiny
  - jar
  - jpeg
  - txt: the surefire test reports
  - xml: the surefire test properties
  - xml.xz: the `gcp-repo` p2 metadata files

It's futile to compress `jar`, `xml.xz`, `zip`, `jpeg`, `png`.  We could add `index`.

Reply:
Ah, I didn't know `-z` is for compression.